alibi_detect/od/pytorch/ensemble.py

```diff
@@ -18,93 +17,63 @@ def __init__(self):
         super().__init__()
 
     def transform(self, x: torch.Tensor):
-        return self._transform(x)
-
-    @abstractmethod
-    def _transform(self, x: torch.Tensor):
-        """Applies class transform to `torch.Tensor`
+        """Public transform method.
 
         Parameters
         ----------
         x
             `torch.Tensor` array to be transformed
         """
-        pass
+        raise NotImplementedError()
 
     @torch.no_grad()
     def forward(self, x: torch.Tensor) -> torch.Tensor:
-        return self.transform(x=x)
+        return self.transform(x)
 
 
 class FitMixinTorch(ABC):
     fitted = False
 
-    def __init__(self):
-        """Fit mixin
-
-        Utility class that provides fitted checks for alibi-detect objects that require to be fit before use.
-        """
-        super().__init__()
-
-    def fit(self, x: torch.Tensor) -> FitMixinTorch:
-        self.fitted = True
-        self._fit(x)
-        return self
+    def fit(self, x: torch.Tensor) -> 'FitMixinTorch':
```

Comment:
Should this be `abstractmethod` given that we inherit from `ABC`? Also see above note for `BaseTransformTorch` - make sure we're consistent.

Reply:
Ah, yeah good catch!